x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,374 @@
+----
+====
+
+[[jwt-validation]]
+==== JWT encoding and validation
+JWTs can be parsed into three pieces:
+
+* Header - Provides information about how to validate the token.
+* Claims - Contains data about the calling user or application.
+* Signature - The raw data that's used to validate the token.
+
+[source,text]
+----
+Header: {"typ":"JWT","alg":"HS256"}
+Claims: {"aud":"aud8","sub":"security_test_user","iss":"iss8","exp":4070908800,"iat":946684800}
+Signature: UnnFmsoFKfNmKMsVoDQmKI_3-j95PCaKdgqqau3jPMY
+----
+
+This example illustrates a partial decoding of a JWT. The validity period is
+from 2000 to 2099, inclusive. JWTs typically have a validity period shorter than
+100 years. The signature in this example is deterministic because the header,
+claims, and HMAC key are fixed. 
+
+The supported JWT encoding is JSON Web Signature (JWS). The JWS header and JWS
+are validated using OpenID Connect ID Token validation rules. Some validation is 
+customizable through <<ref-jwt-settings,JWT realm settings>>.
+
+[[jwt-validation-header]]
+===== Header claims
+The header claims indicate the token type and how the token was signed.
+
+`alg`::
+(Required, String) Indicates the algorithm that was used to sign the token, such
+as `HS256`.
+
+`typ`::
+(Optional, String) Indicates the token type, which must be `JWT`.
+
+[[jwt-validation-oidc]]
+===== Payload claims validated by OIDC
+OIDC ID tokens contain several claims, which provide information about the user
+who is issuing the token, and the token itself. The following claims are
+validated using OIDC ID token rules.
+
+NOTE: You can relax validation of any of the time-based claims by setting 
+`allowed_clock_skew`. This value sets the maximum allowed clock skew before
+validating JWTs with respect to their creation (`iat`), not before (`nbf`), and
+expiration times (`exp`).
+
+`aud`::
+(Required, String) Indicates the audience that the ID token is for, expressed as a 
+comma-separated value (CSV). One of the values must be an exact, case-sensitive
+match to any of the CSV values in the `allowed_audiences` setting.
+
+`exp`::
+(Required, integer) Expiration time for the ID token, expressed in UTC
+milliseconds since epoch.
+
+`iat`::
+(Required, integer) Time that the ID token was issued, expressed in UTC
+milliseconds since epoch.
+
+`iss`::
+(Required, String) Denotes the issuer that created the ID token. The value must
+be an exact, case-sensitive match to the value in the `allowed_issuer` setting.
+
+`nbf`::
+(Optional, integer) Indicates the time before which the JWT must not be accepted,
+expressed as UTC milliseconds since epoch.
+
+`auth_time`::
+(Optional, integer) Time when the user authenticated using the ID token,
+expressed as UTC milliseconds since epoch.
+
+[[jwt-validation-es]]
+===== Payload claims validated by {es}
+{es} validates the following OIDC ID token claims.
+
+NOTE: {es} doesn't validate `nonce` claims, but a custom JWT issuer can add a
```

Comment:
This note makes more sense in the OIDC claims section, not the Elasticsearch section.